### ISSUE TITLE
Async log file writes in the main logger

### ISSUE THREAD

**Issue by roeyb1** [Scarlet-Engine-Team-GameDev-McGill/Scarlet-Engine#114]
Should the logger asynchronously output to the log file using something like `std::thread(WriteFunc).detach()`?
